frontend/gateway/gateway.go

```diff
@@ -640,6 +648,40 @@ func (lbf *llbBridgeForwarder) Solve(ctx context.Context, req *pb.SolveRequest)
 	var defaultID string
 
 	lbf.mu.Lock()
+
+	if res.Attestations != nil {
+		attests := map[string]*pb.Attestations{}
+		for k, as := range res.Attestations {
+			for _, a := range as {
```

Comment:
The `as` variable is confusing, I think this would make it easier to read:
```suggestion
		for k := range res.Attestations {
			for _, attestation := range range res.Attestations[k] {
```